R/cmapplot.R

```diff
@@ -17,73 +17,125 @@
 "_PACKAGE"
 
-cmapplot_globals <- new.env(parent = emptyenv())  # An environment for storing any global variables
-
-
-# Default font handling ---------------------------------------------------
-
-# sizes
-cmapplot_globals$fsize <- list(
-  big = 17,
-  reg = 14,
-  sml = 11
-)
-
-if (.Platform$OS.type == "windows") {
-
-  # Check for Whitney
-  all_fonts <- sysfonts::font_files()
-  whitney_fonts <- dplyr::filter(all_fonts,
-    family %in% c("Whitney Medium", "Whitney Book", "Whitney Semibold") & face == "Regular"
+#' cmapplot global variables
+#'
+#' Expose a list of predefined variables for use by the cmapplot package and its
+#' users.
+#'
+#' @export
+cmapplot_globals <- list(
+
+  ## Colors
+  colors = list(
+    blackish = "#222222"
+  ),
+
+  ## Font sizes
+  fsize = list(
+    big = 17,
+    reg = 14,
+    sml = 11
+  ),
+
+  ## Base typefaces -- modified later by .onLoad()
+  font = list(
+    strong = list(family="Arial", face="bold"),
+    regular = list(family="Arial", face="plain"),
+    light = list(family="Arial", face="plain")
+  ),
+  use_whitney = FALSE,
+
+  ## Establish plotting constants in bigpts (1/72 of inch)
+  ## [t] and [f] in comments indicate whether the constant is called in
+  ## [t]heme_cmap() and/or [f]inalize_plot().
+  consts = list(
+    lwd_originline = 1.6, # [t, ] Width of origin lines. This appears to be the
+                          #        minimum for variation between origin lines and
+                          #        other background lines
+    lwd_gridline = 0.3,   # [t, ] Width of grid lines (non-origin)
+    lwd_plotline = 3,     # [ ,f] Width of line graph lines
+    lwd_topline = 2,      # [ ,f] Width of top line
+    margin_topline_t = 5, # [ ,f] Margin between top edge of image and top line
+    margin_title_t = 5,   # [ ,f] Margin between top line and title
+    margin_title_b = 5,   # [ ,f] Margin between title and caption if
+                          #        `caption_valign = "top"`
+    margin_caption_b = 5, # [ ,f] Margin between caption and bottom edge of image
+    margin_legend_t = 5,  # [ ,f] Margin between top line and plotbox
+    margin_legend_i = 8,  # [t,f] Vertical margin between legends (only applies
+                          #        to multilegend plots)
+    margin_legend_b = 10, # [t,f] Margin between legend and plot (within plotbox)
+    margin_plot_b = 5,    # [ ,f] Margin between plotbox and bottom edge of image
+    margin_title_l = 2,   # [ ,f] Margin between left edge of image and title/caption
+    margin_title_r = 10,  # [ ,f] Margin between title/caption and legend/plot
+                          #        (Both title_l and title_r are deducted from `titlewidth`)
+    margin_plot_r = 10,   # [ ,f] Margin between plotbox and right edge of image
+    padding_plot = c(0,10,0,0),
+                          # [t,f] Padding around the entire plot
+    padding_legend = c(0,0,0,0),
+                          # [t,f] Padding for the legend element. The -9.5
```

Comment:
Flagging to update this documentation (about the -9.5) given the new adjustment making it irrelevant.

Reply:
I addressed this in the base branch.